### PR TITLE
Issue CORE-299 | Wrapping routes in auth layer 

### PR DESCRIPTION
**Pull request by LucasStarlingdePaulaSalles**
# Description

Kind: issue

Section: insprd

Summary: Configuring all routes to validae authentication

Issue link: https://inspr.atlassian.net/browse/CORE-299

### CODE REVIEW

File: cmd/insprd/api/controllers/routes_test.go [hunk 1]
```diff
@@ -89,12 +91,13 @@ func TestServer_initRoutes(t *testing.T) {
 			client := ts.Client()
 			for i, statusCodeResult := range tt.want {
 				reqURL := ts.URL + "/" + tt.name
-
+				// text := "{\"scope\":\"scope_1\"}"
+				// bytes.NewBuffer([]byte(text))
```
Comment (franciscobonand) [inline cmd/insprd/api/controllers/routes_test.go]: Remove commented lines

File: cmd/insprd/api/controllers/base.go [hunk 2]
```diff
@@ -25,6 +26,7 @@ type Server struct {
 	Mux           *http.ServeMux
 	MemoryManager memory.Manager
 	op            operators.OperatorInterface
+	auth          auth.Auth
```
Comment (ptcar2009) [inline cmd/insprd/api/controllers/base.go:29]: Are you initializing the auth on the server?

File: cmd/insprd/main.go [hunk 3]
```diff
@@ -8,26 +8,36 @@ import (
...
 		if err != nil {
 			panic(err)
 		}
 	} else {
+		authenticator = jwtauth.NewJWTauth(pubKey)
 		memoryManager = tree.GetTreeMemory()
 		operator, err = kafka.NewKafkaOperator(memoryManager)
 		if err != nil {
 			panic(err)
 		}
```
Comment (franciscobonand) [inline cmd/insprd/main.go:39]: @ptcar2009 @nicholaspcr do you guys know why the code in "if" is the same as in "else"?

File: pkg/auth/rsa-from-env.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		return nil, err
+	}
+
+	// Then, we can call CryptoPublicKey() to get the actual crypto.PublicKey
+	pubCrypto := parsedCryptoKey.CryptoPublicKey()
+
+	// Finally, we can convert back to an *rsa.PublicKey
+	pub := pubCrypto.(*rsa.PublicKey)
+	return pub, nil
+}
```
Comment (franciscobonand) [inline pkg/auth/rsa-from-env.go]: Maybe test this

**Review by franciscobonand — APPROVED** [review]

File: cmd/authsvc/secret/main.go [hunk 5]
```diff
@@ -53,22 +53,31 @@ func generatePrivateKey() (*rsa.PrivateKey, error) {
 	return privateKey, nil
 }
 
-// encodePrivateKeyToPEM encodes Private Key from RSA to PEM format
-func encodePrivateKeyToPEM(privateKey *rsa.PrivateKey) []byte {
+func encodeKeysToPEM(privateKey *rsa.PrivateKey) ([]byte, []byte, error) {
```
Comment (ptcar2009) [inline cmd/authsvc/secret/main.go]: Change the return to be named
```go
func encodeKeysToPEM(privateKey *rsa.PrivateKey) (pubKey []byte, privKey []byte, err error)
```

**Review by ptcar2009 — APPROVED** [review]

**Pull request merged**